### ISSUE TITLE
Conflict with barryvdh/laravel-ide-helper

### ISSUE THREAD

**Issue by mortenscheel** [lorisleiva/laravel-actions#142]
Hi Loris,

It's been a while :) I like what you've done with this package since last we spoke.
I just wanted to give you a heads up about an issue I discovered, which happens when you run `ide-helper:meta` in a project that has `laravel-actions` installed.
I've posted an issue (https://github.com/barryvdh/laravel-ide-helper/issues/1269) on the ide-helper repo which goes into more detail, but the basic issue is this:

Ide-helper's MetaCommand registers a custom autoloader, which simply just throws an exception. So if none of the "normal" autoloaders can find the class/abstract, an exception gets thrown, and MetaCommand gives up trying to analyze what that abstract resolves to. In your service provider you register a `beforeResolving` callback, which checks `class_exists($abstract)`. And since a lot of the abstracts are just strings like `config`, `request` or `view`, none of the autoloaders can find the class, so an exception gets thrown, and MetaCommand skips that abstract.

I hope this can be fixed on the ide-helper side of things, because I don't think that there's anything wrong with what your package is doing. If you want to fix it on your side, you could wrap the body of the `beforeResolving` callback in a try-catch.

Here's the output of the `ide-helper:meta` command in the project I'm currently working on:
```
Cannot make 'Faker\Generator': Class 'Faker\Provider\en_US\Barcode' not found.
Cannot make 'Illuminate\Auth\Middleware\RequirePassword': Class 'view' not found.
Cannot make 'Illuminate\Contracts\Auth\Authenticatable': Class 'hash' not found.
Cannot make 'Illuminate\Contracts\Routing\ResponseFactory': Class 'view' not found.
Cannot make 'Illuminate\Mail\Markdown': Class 'view' not found.
Cannot make 'agent': Class 'agent' not found.
Cannot make 'auth.driver': Class 'hash' not found.
Cannot make 'auth.password.broker': Class 'hash' not found.
Cannot make 'cookie': Class 'cookie' not found.
Cannot make 'datatables': Class 'datatables' not found.
Cannot make 'datatables.html': Class 'view' not found.
Cannot make 'embed': Class 'embed' not found.
Cannot make 'encrypter': Class 'encrypter' not found.
Cannot make 'excel': Class 'excel' not found.
Cannot make 'form': Class 'form' not found.
Cannot make 'gravatar': Class 'gravatar' not found.
Cannot make 'hash': Class 'hash' not found.
Cannot make 'hash.driver': Class 'hash' not found.
Cannot make 'html': Class 'html' not found.
Cannot make 'mailer': Class 'mailer' not found.
Cannot make 'redirect': Class 'redirect' not found.
Cannot make 'url': Class 'url' not found.
Cannot make 'view': Class 'view' not found.
```

**Comment by lorisleiva** [comment]
Hi Morten 👋 

Sorry for the late reply. Sure I can add a try/catch around `class_exists` if that helps. What exception should I catch?

Also, I'm not familiar with the laravel-ide-helper package so could you just explain to me why it makes `class_exists` throw exceptions?

**Comment by mortenscheel** [comment]
Hi Loris,

The ide-helper package contains a `MetaCommand` artisan command, which creates a `.phpstorm.meta.php` file in the project root. This file tells PhpStorm what all the different abstracts in the container resolve to (e.g. `app('router')` returns an instance of `Illuminate\Routing\Router`).

It throws a `\ReflectionException` ([link](https://github.com/barryvdh/laravel-ide-helper/blob/ebea96213d25bee3ed90da265a2f07c919e40620/src/Console/MetaCommand.php#L144-L156)) if anyone tries to resolve a class that doesn't exist, so if you could catch that, the problem would be solved.

**Comment by lorisleiva** [comment]
Hey @mortenscheel 👋

Sorry about the late reply, I finally took the time to fix this in #164.

Let me know if that all works for you. I've not tagged a new release yet so could you try using `dev-main`?

**Comment by mortenscheel** [comment]
Thanks @lorisleiva .
I've actually stopped using laravel-ide-helper in my projects because I discovered that the laravel-idea plugin for PhpStorm does a much better job.
